lib/Alchemy/Phrasea/Authentication/Provider/AbstractProvider.php

```diff
@@ -19,8 +19,43 @@ abstract class AbstractProvider implements ProviderInterface
 {
     protected $generator;
     protected $session;
+    protected $id;
+    protected $display;
+    protected $title;
+
+    protected function __construct(UrlGenerator $generator, SessionInterface $session, $id, $display, $title)
+    {
+        $this->generator = $generator;
+        $this->session = $session;
+        $this->id = $id;
+        $this->display = $display;
+        $this->title = $title;
+    }
```

Comment:
It may be easier to group `$display` and `$title` into `$options`:
```suggestion
    protected function __construct(UrlGenerator $generator, SessionInterface $session, array $options = [])
    {
        $this->generator = $generator;
        $this->session = $session;
        $this->options = $options;
    }
```

`$options` can also include specific provider options while considering `title` and `display` are reserved keywords.

Reply:
I had the same idea, but "id", "display"... are mandatory args for all providers, as "options" is provider dependant. Passing mandatory args ensure at compile-time that a provider is "well" coded, easier than doing "array_key_exists" at factory-time.

Reply:
OK